### PR TITLE
feat: add create wallet drawer

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
https://github.com/user-attachments/assets/570192ba-3e2d-4d6c-9736-b24aa8cf9e0e

## Summary by CodeRabbit

- **New Features**
	- Introduced a new modal for wallet creation in the Developer Console, enhancing user interactivity.
	- Added the `AddWalletModal` component for a visually appealing wallet management interface, featuring options to create, restore, and view wallet functionalities.

- **Improvements**
	- Enhanced the `Modal` component with support for animated positioning and improved styling for a cohesive design.

- **Bug Fixes**
	- Addressed consistency issues in the modal's default behavior and animations for better user experience.

### CODE REVIEW

File: apps/mobile/src/components/add-wallet/add-wallet-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  Icon,
+  title,
+  subtitle,
+  onPress,
+  isFeatureUnavailable,
+}: AddWalletListItemProps) {
+  const theme = useTheme<Theme>();
+  return (
+    <TouchableOpacity onPress={onPress} py="3" flexDirection="row" gap="4" alignItems="center">
+      {!Icon ? null : (
```
Comment (pete-watters) [inline apps/mobile/src/components/add-wallet/add-wallet-modal.tsx:43]: Can this just be `Icon && (<Box`?

File: apps/mobile/src/components/add-wallet/add-wallet-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                : theme.colors['base.ink.text-primary']
+            }
+            width={24}
+            height={24}
+          />
+        </Box>
+      )}
+      <Box flexDirection="column">
+        <Text variant="label02">{title}</Text>
+        {subtitle ? (
```
Comment (pete-watters) [inline apps/mobile/src/components/add-wallet/add-wallet-modal.tsx:58]: Maybe `&&` here too?

**Review by pete-watters — APPROVED** [review]
Great stuff! 🚀

**Review by kyranjamie — APPROVED** [review]
Looks great @edgarkhanzadian 

Only improvement I'd suggestion would be to compose this UI of two component: one to handle business logic and content, another to encapsulate all the animation logic (of which there is lots).

**Pull request merged**